### DISCUSSION TITLE
Add events to the root elemnt instead of body

### DISCUSSION THREAD

**Discussion by madalinul** [igordanchenko/yet-another-react-lightbox#67] [Q&A] [answered]
Is there a way to handle the drag or wheel events on the root element instead of the document body? I am opening a dialog with input over the lighbox and when I want to select some text with the mouse the drag event is triggering the next slide.

**Comment by igordanchenko** [comment, marked as answer]
The lightbox does not attach pointer and wheel event listeners to the document body. All event listeners are attached to the `yarl__container` div.

<img width="943" alt="Screenshot" src="https://user-images.githubusercontent.com/64441155/216840727-80119191-6565-4fdb-9f8f-53f1b161894b.png">

Have you tried cancelling event propagation on the events you do not want to be handled by the lightbox?

**Reply by madalinul** [reply]
![image](https://user-images.githubusercontent.com/6424322/216845555-d405abe7-5ca0-415f-b2a2-be4185eae8be.png)

It appears that body.yarl__no-scroll is to the body

**Reply by igordanchenko** [reply]
`yarl__no_scroll` is a CSS class attached to the document body, and has nothing to do with event handling. YARL is not using "drag" events, so I'm not clear what point you are trying to make.

**Reply by madalinul** [reply]
https://codesandbox.io/s/yet-another-react-lightbox-examples-forked-s8npjj?file=/src/examples/BasicExample.js&initialpath=/examples/basic

I try to demonstrate in this example, if I open a popver or dialog over the lightbox the events in these elements it triggering the events on the lightbox.
Try to drag and select the text in the dialog.

**Reply by igordanchenko** [reply]
Thank you for providing an example. A picture is worth a thousand words.

Your Popover and Dialog components are descendants of the lightbox in the React tree; hence the events bubble up to the lightbox regardless of where Popover and Dialog components are mounted in DOM.

https://reactjs.org/docs/portals.html#event-bubbling-through-portals

As I mentioned before, you can prevent this behavior by cancelling event propagation.

```jsx
const stopPropagation = (event) => event.stopPropagation();

const eventHandlers = {
  onPointerDown: stopPropagation,
  onPointerMove: stopPropagation,
  onPointerUp: stopPropagation,
  onPointerLeave: stopPropagation,
  onPointerCancel: stopPropagation,
  onKeyDown: stopPropagation,
  onKeyUp: stopPropagation,
  onWheel: stopPropagation
};

// ...

<Popover
  // ...
  {...eventHandlers}
/>

// ...

<Dialog
  // ...
  {...eventHandlers}
/>
```

Here is a code sandbox with a fix for this issue - https://codesandbox.io/s/yet-another-react-lightbox-examples-forked-z2r51d?file=/src/components/ToolbarButtons.js

Another option to address this issue would be to mount Popover and Dialog components higher in the React tree hierarchy (outside of the lightbox component).

**Reply by madalinul** [reply]
Thank you so much, this works.
